sources/client/representation/qt/gaminglineedit.cpp

```diff
@@ -28,8 +26,39 @@ void GamingLineEdit::keyPressEvent(QKeyEvent *event)
     }
     if (event->key() == Qt::Key_F2)
     {
-        setText("OOC ");
+        ToggleOOC();
         return;
     }
     QLineEdit::keyPressEvent(event);
 }
+void GamingLineEdit::ToggleOOC(){
+    QString current_text = text();
+    if(IsOOCMessage(current_text))
+    {
+        if(current_text.length()>3)
```

Comment:
Well, it will work too, but now it looks somewhat ugly. Why not just
```
setText(current_text.mid(3).trim());
```
instead of that `if` and below? We know for sure that `current_text` length is at least 3, so we can safely call `mid(3)` and after that we only need to remove extra spaces.
Just now it looks a little bit too perplexing, can be done much easier.